### PR TITLE
improve rewriteAttrsAction speed [AS-803]

### PR DESCRIPTION
**Pull request by davidangb**
When working with large entities/attributes, the `rewriteAttrsAction` had a perf bottleneck in its comparison of attributes provided by the user and attributes pre-existing in the db. This PR significantly improves performance in this case.

The bottleneck was my bad - it was introduced in #1425. I believe the bottleneck only applies to entities with long strings; so that PR helped in some cases but made it worse in others.

Using the "performance test" included in this PR, I measured before and after the changes in this PR. There are three use cases in the test:
* starting with 0 entities, load 2304 entities with 59 attributes each. This is the "load" test.
* after those entities load, send a batchUpsert of those same 2304 entities/59 attributes, with only one entity/attribute containing a changed value. This mimics downloading a TSV, changing one value, and re-uploading it. This is the "change" test.
* delete all 2304 entities. This is the "delete" test.

Repeating the above tests over 20 iterations, I saw:

&nbsp; | after, ms | before, ms
-- | -- | --
load mean | 3970 | 3653
load min | 3180 | 3059
load max | 5317 | 4483
load sttdev | 492 | 404
change mean | **2934** | **308287**
change min | **2094** | **181005**
change max | **4878** | **345957**
change stddev | **678** | **42672**
delete mean | 6947 | 6304
delete min | 6333 | 5893
delete max | 7521 | 7108
delete stddev | 370 | 308

The tests at the bottom of `AttributeComponentSpec` validate this PR for correctness.

Reviewer: this PR contains both the important runtime fix and the "perf" test to validate it. The perf test must be run manually, it is not by default part of the unit test suite. If you have any objections to the perf test, I would happily split that off into a separate PR and just merge the runtime fix.

### CODE REVIEW

File: project/Dependencies.scala
Comment (davidangb) [inline project/Dependencies.scala:92]: introduced - to test code only - to help calculate stddev/mean. There are other ways to do that, this seemed simplest.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala
Comment (davidangb) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala:398]: 1) let Scala compare case classes instead of manually inspecting the fields of the `RECORD` class
2) use `Set.contains` to look for existing attributes instead of `Traversable.exists`

**Review by MatthewBemis — APPROVED** [review]
LGTM. I read the test but did not run it myself because a) I trust the results you posted, and b) the actual code change itself is what I care the most about

**Comment by davidangb** [comment]
merging with one thumb instead of two as an override; will get a second review soon after this is released as a hotfix.

**Pull request merged**
